src/azul/openapi/schema.py

```diff
@@ -117,6 +122,90 @@ def array(item: TYPE, *items: TYPE, **kwargs):
     return array_type(make_type(item), *map(make_type, items), **kwargs)
 
 
+def enum_type(items: Iterable) -> JSON:
+    """
+    Specify allowed values for a parameter schema.
+
+    >>> from azul.doctests import assert_json
+    >>> assert_json(enum_type({1, 2}))
+    {
+        "type": "integer",
+        "format": "int64",
+        "enum": [
+            1,
+            2
+        ]
+    }
+
+    >>> from azul.doctests import assert_json
+    >>> assert_json(enum_type({'foo', 'bar'}))
+    {
+        "type": "string",
+        "enum": [
+            "bar",
+            "foo"
+        ]
+    }
+    """
+    items = sorted(items)
+    type_ = one(set(map(type, items)))
+    return dict(make_type(type_), enum=items)
+
+
+def path_param(name: str, type_: TYPE, description: Optional[str] = None) -> JSON:
```

Comment:
The param stuff should be in a separate `params` module. `schema` is for the JSONSchema subset of OpenAPI.

Instead of `schema.path_param(…)` an invocation would be `params.path` which is a bit shorter.